libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java

```diff
@@ -67,4 +79,16 @@ public void setFeatureSetFromList(List<String> featureSet) {
         this.featureSet =
                 (featureSet != null && !featureSet.isEmpty()) ? String.join(",", featureSet) : null;
     }
+
+    public void pushState(IpvJourneyTypes journeyType, String state) {
+        this.stateStack.push(new JourneyState(journeyType, state));
+    }
+
+    public void pushState(String state) throws NoCurrentStateException {
+        var currentState = this.stateStack.peek();
+        if (currentState == null) {
+            throw new NoCurrentStateException();
```

Comment:
I did think about this. But wondered about handling it gracefully were it to ever happen, rather than just propergating it to the runtime. To avoid that we'd need to catch it. And at that point we might as well have a checked...

Reply:
Fair enough, I think it basically depends whether you think it's worth handling gracefully (given that either way it'll produce an error for the user).